### PR TITLE
Refactor exec/expression into exec/call and exec/new

### PR DESCRIPTION
**Pull request by croraf**
Minor refactor. Also added BoaProfiled to `impl Executable for New`

### CODE REVIEW

File: boa/src/exec/new/mod.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    BoaProfiler,
+};
+
+impl Executable for New {
+    fn run(&self, interpreter: &mut Interpreter) -> ResultValue {
+        let _timer = BoaProfiler::global().start_event("New", "exec");
+        // let (callee, args) = match call.as_ref() {
+        //     Node::Call(callee, args) => (callee, args),
+        //     _ => unreachable!("Node::New(ref call): 'call' must only be Node::Call type."),
+        // };
```
Comment (Razican) [inline boa/src/exec/new/mod.rs:17]: Can this be removed now? I think a `New` node should always have a `Call`, so it's checked at the type level, right?

Comment (croraf) [reply boa/src/exec/new/mod.rs:17]: I don't understand your comment. Do you mean to remove the comments and leave the timer? 

The comments were here before my work, if you want me to remove them I will?

Comment (Razican) [reply boa/src/exec/new/mod.rs:17]: Yes, sorry, I selected the timer too, but I meant the comments. a `New` node will now always have a `Call`, so no need to check for it.

I was wondering, though, should we move the New/Call AST nodes too from the expression module in `syntax::ast`?

Comment (croraf) [reply boa/src/exec/new/mod.rs:17]: I think this would be good, yes.

Comment (croraf) [reply boa/src/exec/new/mod.rs:17]: Check PR #542

**Pull request merged**